### DISCUSSION TITLE
Add some components: min, max, and edge detector

### DISCUSSION THREAD

**Discussion by Orum** [FakeFishGames/Barotrauma#8533] [Ideas and suggestions]
Minimum and maximum components would be nice to have, i.e. something that outputs the lesser/greater of two inputs respectively.

Additionally, an edge detector, with options for triggering on rising, falling, or both edges of an input signal, would be appreciated.

**Comment by gerakis** [comment]
I also want an edge detector.

**Comment by mygamingaccount** [comment]
Minimum/maximum should be able to select the "lowest / highest value in the previous timeframe" so you could send multiple signals to it

**Comment by FalcoGer** [comment]
Edge detector would be nice, that gives the new signal as output whenever it changes. Ie "Hello" to "world" would output "world" for one frame only. Also agreed, min/max is a pain to do right now.

**Reply by mygamingaccount** [reply]
There's a very simple solution that does exactly this #6470

**Reply by gerakis** [reply]
Is this being added to the game?

**Reply by mygamingaccount** [reply]
I don't think so.

**Comment by gerakis** [comment]
I discovered you can make either a rising edge or falling edge detector by using a signal component tied to a delay component set to "reset when same signal"

We would still benefit from a proper component for this though. If nothing else it would make wiring diagrams more readable.
